core/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -194,9 +194,18 @@ protected void doStop() {
                 if (holderToNotify != null) {
                     // callback that an exception happened, but on a different thread since we don't
                     // want handlers to worry about stack overflows
-                    threadPool.generic().execute(new Runnable() {
+                    threadPool.generic().execute(new AbstractRunnable() {
                         @Override
-                        public void run() {
+                        public void onRejection(Throwable t) {
+                            // if we get rejected during node shutdown we don't wanna bubble it up
+                            logger.debug("failed to notify response handler on rejection", t);
```

Comment:
can we also log the action name and the fact that we're shutting down? Also, I think this can be trace? (it will flood the logs ...)